datasets/inception_subset.py

```diff
@@ -0,0 +1,174 @@
+        return self.num_plants
+
+    def __get_file_date(self, file, plant, mod):
+        file_format = f"{self.root_dir}/{self.experiment}_{mod}_{plant}%Y-%m-%d_%H:%M:%S"
+        return datetime.strptime(file, file_format).date()
+
+    # TODO: this one did not work because getitem gets only one parameter (plant)
+    # @staticmethod
+    # def __plant_inception__(self, plant, mod):
+    #     tensors = []
+    #     data_list = self.plant_files_dict[mod][plant]
+    #     if len(data_list) > 0:
+    #         plant_name = f'plant_{plant}_'
+    #         print(plant_name, mod)
+    #         for file in data_list:
+    #             date_is = self.__get_file_date__(file, plant_name)  # file date
+    #             if self.start_date <= date_is <= self.end_date:
+    #                 inception_embedding = torch.load(file)
+    #                 tensors.append(inception_embedding)
+    #     plant_inception = torch.stack([tensor for tensor in tensors])
+    #
+    #     return plant_inception
+
+    def __plant_inception_one_mod(self, plant, mod):
+        tensors = []
+        data_list = self.plant_files_dict[mod][plant]
+        # if len(data_list) > 0:
+        if data_list is not None:
+            plant_name = f'plant_{plant}_'
+            for file in data_list:
+                date_is = self.__get_file_date(file, plant_name, mod)  # file date
+                if self.start_date <= date_is <= self.end_date:
+                    inception_embedding = torch.load(file)
+                    tensors.append(inception_embedding)
+            plant_inception_mod = torch.stack([tensor for tensor in tensors])
+            return plant_inception_mod
+        else:
+            print('at least one of the mods is empty, check experiment mods!!!')
+            return None
+
+    def __plant_inception_all_mods(self, plant):
+        plant_inception_all_mods = {}
+        for mod in parameters.used_modalities:
+            plant_inception_all_mods[mod] = self.__plant_inception_one_mod(plant, mod)
+        return plant_inception_all_mods
+
+    def __getitem__(self, idx):
+        plant = idx % self.num_plants
+        sample = dict(
+            inception=self.__plant_inception_all_mods(plant),
+            label=labels[self.experiment][plant],
+            plant=plant)
+
+        return sample
+
+
+class InceptionSubset(Dataset):
+    def __init__(self, used_plants, experiment, root_dir, used_modalities, plants: List[int]):
+        self.labels = labels
+        self.experiment = experiment
+        self.root_dir = root_dir
+        self.modalities = used_modalities
+        self.data = used_plants
+        self.plants = plants
+        self.num_plants = len(plants)
+
+        print('self.plants  = ', self.plants)
+
+    def __len__(self):   # needs to be corrected to the number of items per plant for each modality
+        return self.num_plants
+
+    def __getitem__(self, idx):
+        plant = self.plants[idx % self.num_plants]
```

Comment:
- [ ] rename to something a bit more specific
```suggestion
        plant_num = self.plants[idx % self.num_plants]
```